packages/server/modules/core/tests/favoriteStreams.spec.js

```diff
@@ -0,0 +1,294 @@
+/* instanbul ignore file */
```

Comment:
Event though tests for this functionality could've been put inside `streams.spec.js`, I put everything in a separate file for 2 reasons:
1) Tests in streams.spec.js aren't very deterministic, because there are tests depending on the results of other tests. This makes them flaky depending on the order they're ran. So I wanted to create a new one with the test state only being prepared in `before` calls and tested in a way that its not not mutated in a serious way accross tests.
2) I wanted to organize tests in such a way that there's more re-use of test states set up in `before` calls in nested `describe` blocks, so that there's less duplication of test state/data creation code. Jest suggests devs to write the titles of `describe` and `it` calls in such a way that they make a grammatically correct sentence at the end, because organizing `describe`/`it` blocks in this way is more conducive to well organized & nested tests and it's easier to understand what's going on that way. So instead of a test being named something like 'Objects > Repository > error when object deleted' they would end up being named 'Favorite streams > when authenticated > and favorited > can be unfavorited'.